mocksipipeline/instrument/optics/aperture.py

```diff
@@ -38,7 +44,52 @@ def __init__(self, diameter: u.cm, center_to_center_distance: u.cm):
     @property
     @u.quantity_input
     def area(self) -> u.cm ** 2:
-        return np.pi * (self.diameter/2) ** 2 + self.diameter * self.center_to_center_distance
+        return np.pi * (self.diameter / 2) ** 2 + self.diameter * self.center_to_center_distance
+
+    def mask(self, oversample=8):
+        pinhole_diameter = self.diameter
+        center2center_distance = self.center_to_center_distance
+
+        aperture_size = (center2center_distance + pinhole_diameter) * 1.05
+        x = np.linspace(-aperture_size / 2, aperture_size / 2, num=int((aperture_size * oversample).value),
+                        endpoint=True)
+        x, y = np.meshgrid(x, x)
+
+        # shift coordinate system
+        x_shift = x - center2center_distance / 2
+
+        r = np.sqrt(x_shift * x_shift + y * y)
+        pinhole = np.ones_like(r).value
+        pinhole[r < pinhole_diameter / 2] = 0
+
+        # shift coordinate system
+        x_shift = x + center2center_distance / 2
+
+        r = np.sqrt(x_shift * x_shift + y * y)
+        pinhole2 = np.ones_like(r).value
+        pinhole2[r < pinhole_diameter / 2] = 0
+
+        x_cut = np.ones_like(r).value
+        y_cut = np.ones_like(r).value
+
+        x_cut *= x > -center2center_distance / 2
+        x_cut *= x < center2center_distance / 2
+
+        y_cut *= y > -pinhole_diameter / 2
+        y_cut *= y < pinhole_diameter / 2
+
+        rectangle = x_cut * y_cut
+        rectangle = 0 ** rectangle
```

Comment:
I don't understand. Why is `rectangle` just all zeros? What's the purpose of this?